src/sagas/pushNotification.js

```diff
@@ -0,0 +1,412 @@
+/**
+ * this is the message key for localization of new transaction title
+ */
+const NEW_TRANSACTION_RECEIVED_TITLE = 'new_transaction_received_title';
+/**
+ * this is the channel id for the transaction notification
+ */
+const TRANSACTION_CHANNEL_ID = 'transaction';
+const TRANSACTION_CHANNEL_NAME = t`Transaction`;
+
+const getDeviceId = async () => {
+  try {
+    const deviceId = await messaging().getToken();
+    return deviceId;
+  } catch (error) {
+    console.error(`Error getting deviceId: ${error.message}`, error);
+    return null;
+  }
+};
+
+/**
+ * localization utils to map the message key to the correct message to localize
+ */
+const localization = {
+  keys: new Set([
+    NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS,
+    NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS,
+    NEW_TRANSACTION_RECEIVED_TITLE
+  ]),
+  hasKey: (key) => localization.keys.has(key),
+  getMessage: (key, args) => {
+    if (!localization.hasKey(key)) {
+      console.log('unknown key', key);
+      return '';
+    }
+
+    let message = '';
+    if (key === NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS) {
+      if (!args) {
+        console.log('args is null for key', key);
+        return '';
+      }
+      const countArgs = args.length;
+      if (countArgs === 3) {
+        const [firstToken, secondToken, other] = args;
+        const otherCount = parseInt(other, 10);
+        message = t`you have received ${firstToken}, ${secondToken} and ${otherCount} other token on a new transaction.`;
+      } else if (countArgs === 2) {
+        const [firstToken, secondToken] = args;
+        message = t`you have received ${firstToken} and ${secondToken}.`;
+      } else if (countArgs === 1) {
+        const [firstToken] = args;
+        message = `you have received ${firstToken}.`;
+      }
+    } else if (key === NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS) {
+      message = t`There is a new transaction in your wallet.`;
+    } else if (key === NEW_TRANSACTION_RECEIVED_TITLE) {
+      message = t`New transaction received`;
+    }
+    return message;
+  }
+};
+
+const onForegroundMessage = async (message) => {
+  await messageHandler(message);
+};
+
+const onBackgroundMessage = async (message) => {
+  await messageHandler(message);
+};
+
+/**
+ * Handle the message received when application is in foreground and background (not closed) state
+ * @param {{ data: Object, from: string, messageId: string, sentTime: number, ttl: number }} message - Message received from wallet-service
+ * @example
+ * {
+ *   bodyLocArgs: '[\"10 T2\",\"5 T1\",\"2\"]',
+ *   bodyLocKey: 'new_transaction_received_description_with_tokens',
+ *   titleLocKey: 'new_transaction_received_title',
+ *   txId: 'txId1',
+ * }
+ * @inner
+ */
+const messageHandler = async (message) => {
+  const { data } = message;
+  if (!localization.hasKey(data.titleLocKey)) {
+    console.log('unknown message titleLocKey', data.titleLocKey);
```

Comment:
Remove comment?

Reply:
Show we avoid to log?

Reply:
I guess it's OK to log, the problem is that it's going nowhere, so it's kind of pointless I guess